packages/wallet/src/hdkey.ts

```diff
@@ -1,18 +1,22 @@
 import { HDKey } from 'ethereum-cryptography/hdkey.js'
-
+import { mnemonicToSeedSync } from 'ethereum-cryptography/bip39.js'
 import { Wallet } from './wallet.js'
 
 export class EthereumHDKey {
   /**
    * Creates an instance based on a seed.
-   *
-   * For the seed we suggest to use [bip39](https://npmjs.org/package/bip39) to
-   * create one from a BIP39 mnemonic.
    */
   public static fromMasterSeed(seedBuffer: Uint8Array): EthereumHDKey {
     return new EthereumHDKey(HDKey.fromMasterSeed(seedBuffer))
   }
 
+  /**
+   * Creates an instance based on BIP39 mnemonic phrases
+   */
+  public static fromMnemonic(mnemonic: string, passphrase?: string): EthereumHDKey {
+    return EthereumHDKey.fromMasterSeed(Buffer.from(mnemonicToSeedSync(mnemonic, passphrase)))
```

Comment:
Remove `Buffer.from`, it should always consume Uint8Array, which it already returns

Reply:
Yes, just did it. Thank you.